go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -252,8 +257,8 @@ func (oc *Controller) updateEgressFirewall(oldEgressFirewall, newEgressFirewall
 
 	ipv4ASHashName, ipv6ASHashName := addressSet.GetASHashNames()
 	match := generateMatch(ipv4ASHashName, ipv6ASHashName, []matchTarget{
-		{matchKindV4CIDR, "0.0.0.0/0"},
-		{matchKindV6CIDR, "::/0"},
+		{matchKindV4CIDR, unknownIPv4Destination},
```

Comment:
Good question, @JacobTanenbaum any input?

Reply:
as part of updating egressfirewalls before modifying the egressfirewall rules themselfs a blanket deny all rule is created for the namespace to remove the case where external traffic that should be denied is allowed for a small amount of time. this call to `generateMatch()` is creating the blanket deny rule for the update

Reply:
but couldn't this blanket deny rule accidentally block traffic to egress during the update that shouldn't be blocked? I guess we will need to talk more in depth on this tmrw. I'll ping you and Alex and we can discuss.

Reply:
This is only done for a namespace which already had an egress firewall object assigned. It should thus only block that namespace if I am reading this right.

Reply:
It is more secure to update this way, during update we maintain any deny rules by dening all the traffic. there are only a  few ways to update the set of rules since we can only update one at a time.
1. remove all the rules and apply the new ones - in this case traffic that was previously denied in the cluster has a small window to enter the cluster
2. make a deny all rule while updating - For a period of time we are blocking legitimate traffic but at no time is traffic that was thought to be blocked allowed to leave the cluster
3. remove and add one at a time - the behavior is inconsistent and dependent on the rules themselves. a rule could be swapped from a deny to allow or the rules could be completely reordered.

There is potential to make the update smarter but I think it would be a lot of calculation for a small gain.

Reply:
As discussed on slack we should batch the commands into a single nbctl transaction to avoid any window like this.

Reply:
Covered by https://github.com/ovn-org/ovn-kubernetes/pull/2223